pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -124,19 +126,52 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 
 	sts, err := buildStatefulSet(mdb)
 	if err != nil {
-		log.Warnf("error building StatefulSet: %s", err)
+		log.Infof("Error building StatefulSet: %s", err)
 		return reconcile.Result{}, nil
 	}
 
 	if err = r.client.CreateOrUpdate(&sts); err != nil {
-		log.Warnf("error creating/updating StatefulSet: %s", err)
+		log.Infof("Error creating/updating StatefulSet: %s", err)
 		return reconcile.Result{}, err
+	} else {
+		log.Infof("StatefulSet successfully Created/Updated")
+	}
+
+	log.Infof("waiting for StatefulSet %s/%s to reach ready state", mdb.Namespace, mdb.Name)
+	set := appsv1.StatefulSet{}
+	timedOut, err := r.client.WaitForCondition(types.NamespacedName{Name: mdb.Name, Namespace: mdb.Namespace}, time.Second*3, time.Second*30, &set, func() bool {
+		return statefulset.IsReady(set)
+	})
+
+	if timedOut {
+		log.Infof("Stateful Set has not yet reached the ready state, requeuing reconciliation")
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	if err != nil {
+		log.Errorf("error polling for statefulset: %+v", err)
+		return reconcile.Result{}, err
+	}
+
+	log.Infof("Stateful Set reached ready state!")
+
+	if err := r.updateStatusSuccess(&mdb); err != nil {
+		log.Infof("Error updating the status of the MongoDB resource: %+v", err)
+		return reconcile.Result{}, nil
 	}
 
 	log.Info("Successfully finished reconciliation", "MongoDB.Spec:", mdb.Spec, "MongoDB.Status", mdb.Status)
 	return reconcile.Result{}, nil
 }
 
+func (r ReplicaSetReconciler) updateStatusSuccess(mdb *mdbv1.MongoDB) error {
+	mdb.UpdateSuccess()
+	if err := r.client.Status().Update(context.TODO(), mdb); err != nil {
```

Comment:
Will this mean that the object will go to error state if the Status object was not updated?

Reply:
No, at the moment there's no error state for the resource